From bce1227d9089396f4fe44d2e726fc0af6e08a657 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Wed, 1 Aug 2018 09:16:34 +0800
Subject: [PATCH] media: rk-isp10: fix fmt select

if enumerate all the sensor fmt, don't find matching format
then should be forced to using the last one.

Change-Id: I15949ed66f583019a522ee56844e57813c6f6e49
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rk-isp10/cif_isp10.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/rk-isp10/cif_isp10.c b/drivers/media/platform/rk-isp10/cif_isp10.c
index 507c954cc81b..de2539e6d78d 100644
--- a/drivers/media/platform/rk-isp10/cif_isp10.c
+++ b/drivers/media/platform/rk-isp10/cif_isp10.c
@@ -1215,7 +1215,7 @@ static int cif_isp10_img_src_select_strm_fmt(
 				matching_format_found = true;
 			}
 		// FIXME::GST set fomat(any@32768x32768) failed, force pass
-		} else {
+		} else if (!matching_format_found) {
 			request_strm_fmt.frm_fmt.width =
 				strm_fmt_desc.min_frmsize.width;
 			request_strm_fmt.frm_fmt.height =
-- 
2.35.3

